File: StrategyPattern/java_/Computer/Computer.java
package StrategyPattern.java_.Computer;
import StrategyPattern.java_.Computer.CPU.*;
import StrategyPattern.java_.Computer.GPU.*;

public class Computer {
    private CPU cpu;
    private GPU gpu1;
    private GPU gpu2;

    public Computer() {
        this.cpu = null;
        this.gpu1 = null;
        this.gpu2 = null;
    }

    public CPU getCpu() {
        return this.cpu;
    }

    public void setCpu(CPU cpu) {
        this.cpu = cpu;
    }

    public GPU getGpu1() {
        return this.gpu1;
    }

    public void setGpu1(GPU gpu1) {
        this.gpu1 = gpu1;
    }

    public GPU getGpu2() {
        return this.gpu2;
    }

    public void setGpu2(GPU gpu2) {
        this.gpu2 = gpu2;
    }

    public void showInfo(){
        System.out.println("The component of this computer");
        System.out.println(String.format("CPU   : %s", cpu.showInfo()));
        System.out.println(String.format("GPU 1 : %s", gpu1 == null ? "None" : gpu1.showInfo()));
        System.out.println(String.format("GPU 2 : %s", gpu2 == null ? "None" : gpu2.showInfo()));
    }
}

File: AdapterPattern/cpp/src/computer.cpp
#include <iostream>
#include "header/computer.h"

using namespace std;

string AMD_Ryzen7_3700X::showInfo(){
    return "AMD Ryzen7 3700X";
}

string Old_CPU_Adapter::showInfo(){
    return this->getName();
}

string NVIDIA_GeForce_GTX_1660_SUPER::showInfo(){
    return "NVIDIA GeForce GTX 1660 SUPER";
}

shared_ptr<CPU> Computer::getCpu(){
    return cpu;
}

void Computer::setCpu(shared_ptr<CPU> &cpu){
    this->cpu = cpu;
}

shared_ptr<GPU> Computer::getGpu1(){
    return gpu1;
}

void Computer::setGpu1(shared_ptr<GPU> &gpu){
    gpu1 = gpu;
}

shared_ptr<GPU> Computer::getGpu2(){
    return gpu2;
}

void Computer::setGpu2(shared_ptr<GPU> &gpu){
    gpu2 = gpu;
}

void Computer::showInfo(){
    cout << "The component of this computer" << endl;
    cout << "CPU   : " << cpu->showInfo() << endl;
    cout << "GPU 1 : " << (gpu1 == nullptr ? "None" : gpu1->showInfo()) << endl;
    cout << "GPU 2 : " << (gpu2 == nullptr ? "None" : gpu2->showInfo()) << endl;
}